driver-core/src/main/com/mongodb/internal/operation/CommandBatchCursor.java

```diff
@@ -258,6 +261,9 @@ private CommandCursorResult<T> toCommandCursorResult(final ServerAddress serverA
         return commandCursorResult;
     }
 
+    void setCloseImmediately(final boolean closeImmediately) {
```

Comment:
Would be good to explain what his is and why its needed? I dont fully understand the reasoning here.

Reply:
The current implementation is designed to automatically reset the timeoutMS for cursor closure, ensuring that the cursor has sufficient time to release resources. This behavior particularly impacts the `ChangeStreamCursor` which enforces a unified timeout duration for the entire 'next' operation, including resume attempts and the internal process of closing the cursor.

The method modifies this behavior with respect to timeouts. When enabled, it prevents the automatic reset of the cursor's timeout for closing.

I realized that the method name is not reflecting its purpose. Therefore, I have updated the name to `setCloseWithoutTimeoutReset` and added the JavaDoc documentation for clarity.